### PR TITLE
Improve resilience of rabbitmq dispatcher and mongodb initialization

### PR DESCRIPTION
**Pull request by jaardila-3**
web reference health checks: https://medium.com/@jeslurrahman/implementing-health-checks-in-net-8-c3ba10af83c3

### CODE REVIEW

File: src/api/TheFullStackTeam.API/Program.cs [hunk 1]
```diff
@@ -9,96 +15,101 @@
...
-            app.MapControllers();
-            app.Run();
-        }
+            Converters = { new UlidJsonConverter() }
+        });
+
+        builder.Services.AddAuthorization();
+        builder.Services.AddControllers();
+        // Congiguring Health Ckeck
+        builder.Services.AddCustomHealthChecks(builder.Configuration);
```
Comment (juangcarmona) [inline src/api/TheFullStackTeam.API/Program.cs:73]: After adding custom health checks, we should map separate endpoints for liveness and readiness checks in the middleware. This helps differentiate application availability from the readiness to handle incoming requests, which is especially useful in orchestration tools like Kubernetes.

File: src/api/TheFullStackTeam.API/Program.cs [hunk 2]
```diff
@@ -9,96 +15,101 @@
...
+        //});
+
+        app.UseHttpsRedirection();
+        app.UseRouting();
+        app.UseAuthentication();
+        app.UseAuthorization();
+        app.MapControllers();
+
+        // HealthCheck Middleware
+        app.MapHealthChecks("/api/health", new HealthCheckOptions()
```
Comment (juangcarmona) [inline src/api/TheFullStackTeam.API/Program.cs:107]: Consider replacing /api/health with two different endpoints: /health/liveness for basic application health and /health/readiness for critical dependency checks. Use Predicate to filter checks by their tags (liveness or readiness). This approach will improve the application lifecycle management, particularly in cloud deployments.

i.e.:

```
app.MapHealthChecks("/health/liveness", new HealthCheckOptions
{
    Predicate = (check) => check.Tags.Contains("liveness"),
    ResponseWriter = UIResponseWriter.WriteHealthCheckUIResponse
});

app.MapHealthChecks("/health/readiness", new HealthCheckOptions
{
    Predicate = (check) => check.Tags.Contains("readiness"),
    ResponseWriter = UIResponseWriter.WriteHealthCheckUIResponse
});
```
The idea is to Split the /api/health endpoint into two endpoints: /health/liveness for verifying that the app process is running, and /health/readiness for checking the status of dependencies like databases and message brokers. This helps Kubernetes and other orchestrators make better decisions about container restarts or taking the instance out of the load balancer.

**Review by juangcarmona — CHANGES_REQUESTED** [review]
In general looks good but we have inherited (my mistake) some code that looks core, and we might pay special attention to it.

File: src/api/TheFullStackTeam.Infrastructure/Messaging/RabbitMq/RabbitMQEventDispatcher.cs [hunk 3]
```diff
@@ -1,69 +1,104 @@
...
     private readonly RabbitMQSettings _settings;
+    private readonly ILogger<RabbitMQEventDispatcher> _logger;
 
-    public RabbitMQEventDispatcher(IOptions<RabbitMQSettings> settings)
+    public RabbitMQEventDispatcher(IOptions<RabbitMQSettings> settings, ILogger<RabbitMQEventDispatcher> logger)
     {
+        _logger = logger;
         _settings = settings.Value;
-        var factory = new ConnectionFactory()
+        _connection = new Lazy<IConnection>(() =>
```
[the remark this answers is not in the document]
Comment (jaardila-3) [reply src/api/TheFullStackTeam.Infrastructure/Messaging/RabbitMq/RabbitMQEventDispatcher.cs]: resolved

File: src/api/TheFullStackTeam.API/Program.cs (hunk 1 above)
Comment (jaardila-3) [reply src/api/TheFullStackTeam.API/Program.cs:74]: I didn't understand the difference between the previous suggestion and this one.

File: src/api/TheFullStackTeam.API/Program.cs [hunk 4]
```diff
@@ -9,96 +15,101 @@
...
 using TheFullStackTeam.Infrastructure.Persistence.Sql.Services;
 using TheFullStackTeam.Infrastructure.Services;
-using Serilog;
-using System.Text.Json;
-using TheFullStackTeam.Api.Extensions;
-using TheFullStackTeam.Api.Middlewares;
 
-namespace TheFullStackTeam.Api
-{
+namespace TheFullStackTeam.Api;
```
[the remark this answers is not in the document]
Comment (jaardila-3) [reply src/api/TheFullStackTeam.API/Program.cs]: I Added .editorconfig to Solution and I Used File-Scoped Namespace

**Review by juangcarmona — APPROVED** [review]
The only Change I need is to translate those comments into spanish

**Pull request merged**